### ISSUE TITLE
Nested dialogs closeEvent not working as expected

### ISSUE THREAD

**Issue by AlexRoig** [vaadin/vaadin-dialog-flow#80]
Hi, 

When I create a dialog in a dialog the event propagation of close event is not working as intended.
Here are the behaviours observed when the second (nested) dialog is opened on top:
a) When clicking outside the dialog to dismiss it, the server side events registered in Dialog#addDialogCloseActionListener() are executed as expected but the dialog is not closed. You can keep clicking outside and the dialog will not disappear and the events will be executed at each click.
b) When pressing ESC key to close the dialog the events, we have the same behaviour as a).
c) When adding a button to the second dialog and adding a clickListener to the button that calls the Dialog#close() method, then the events registered on Dialog#addDialogCloseActionListener() are not executed.

I'll try to provide a minimum code to test the functionality. Currently tested with RC5

Thx, 
Alex

**Comment by denis-anisimov** [comment]
Thank you for reporting but I cannot reproduce the issue with the latest dialog release.

I'm using the code below
```
Dialog dialog = new Dialog();
        dialog.add(new Text("xxxxxx"));

        Dialog subDialog = new Dialog();
        subDialog.add(new Text("uuuu"));
        dialog.add(subDialog);

        dialog.open();

        NativeButton button = new NativeButton("Open subdialog",
                event -> subDialog.open());
        dialog.add(button);
```

Also the second dialog is closed if it's not added into the first dialog (the line ` dialog.add(subDialog);` is commented out).

Another thing to note is: it looks like you are describing the client side behavior only which means this is the web component behavior, not the server component. It might be that this behavior is fixed in the final version.

I'm closing the ticket since I cannot reproduce it based on the description. If you have a code which shows how to reproduce it please reopen and add this code.

**Comment by denis-anisimov** [comment]
Alright, now I see what's the problem. It's `addDialogCloseActionListener` method which adds some JS on the client side and this causes issues.

**Comment by denis-anisimov** [comment]
This ticket is strongly connected with https://github.com/vaadin/vaadin-dialog-flow/issues/68.
In fact the fix which I'm going to do should fix this issue as well.

There will be a test for the scenario from this ticket.
I'm not closing this as a duplicate since the scenario is not the same but it will be fixed by the same PR.

As a final note I have to say that the name for the `addDialogCloseActionListener` is bad and confusing. I will update javadocs for it. Unfortunately it's too late to change the method name.

Please note that if you use the `addDialogCloseActionListener` method then it's your responsibility to close the dialog. The dialog won't be closed for you automatically if you click outside it. You should call `close` method on the server side in the listener implementation. Otherwise you are using the method wrong.
I you still need automatic closing of the dialog and be notified when it happens then you should use `addOpenedChangeListener` which is called whenever `opened` property value is updated (meaning also when it becomes `false`  which means the dialog is closed).

**Comment by denis-anisimov** [comment]
The PR for this : https://github.com/vaadin/vaadin-dialog-flow/pull/91.

I will close this ticket since it will be fixed along with #68 though the latter ticket is no a duplicate.

Feel free reopen if you still think it's not fixed with the mentioned PR.
Thanks.
